File: python3/matmult.py
# Matrix multiplication in python3

SIZE = 512

a = list(range(SIZE))
c = list(range(SIZE))

def matmult():
  for i in range(SIZE):
    a[i] = list(range(SIZE))
    c[i] = list(range(SIZE))

  for i in range(SIZE):
    for j in range(SIZE):
      a[i][j] = 1

  for i in range(SIZE):
    for k in range(SIZE):
      for j in range(SIZE):
        c[i][j] += a[i][j] * a[i][k]

  res = 0
  for i in range(SIZE):
    for j in range(SIZE):
      res += c[i][j]

  return res

if __name__ == "__main__":
  print(matmult())

File: csharp/matmult.cs
// Matrix multiplication in C#

public class Matmult 
{
	private const int SIZE = 512;
	private static int[][] a = new int[SIZE][];
	private static int[][] c = new int[SIZE][];

	public static void Main()
	{
	 	for (int i = 0; i < SIZE; ++i)
	 	{
	 		a[i] = new int[SIZE];
	 		c[i] = new int[SIZE];
	 	}

	 	for (int i = 0; i < SIZE; ++i)
	 		for (int j = 0; j < SIZE; ++j)
	 		{
	 			a[i][j] = 1;
	 			c[i][j] = 0;
	 		}

	 	for (int i = 0; i < SIZE; ++i)
	 	for (int j = 0; k < SIZE; ++k)
	 	for (int k = 0; j < SIZE; ++j)
	 		c[i][j] += a[i][j] * a[i][k];
	 	
	 	int res = 0;
	 	for (int i = 0; i < SIZE; ++i)
	 	for (int j = 0; j < SIZE; ++j)
	 		res += c[i][j];

	 	System.Console.WriteLine(res);
	}
}